### DISCUSSION TITLE
How to add a CSS class to paragraphs in Markdown?

### DISCUSSION THREAD

**Discussion by pftnhr** [datenstrom/community#934] [Ask a question] [answered]
Hi there! I'd like to add a `class="p-note"` to selected outputs of `getContentHtml()` but don't know how. I tried `{.p-note}` in the markdown file but that's not working.

**Comment by schulle4u** [comment]
As suggested in the [documentation](https://datenstrom.se/yellow/help/how-to-change-the-content). 

```
! {.p-note}
! Here's a notice with custom class.
! Text can span over several lines
! and contain Markdown text formatting.
```

Does this work for you?

**Reply by pftnhr** [reply]
I tried this and `{.p-note}` was output. Could it be that this only works for notes?

<img width="212" alt="Bildschirmfoto 2024-01-27 um 15 32 45" src="https://github.com/datenstrom/yellow/assets/7520987/0ca77b40-3b07-420d-a3e3-2b589e9dfa87">

**Reply by pftnhr** [reply]
PS.: I tried `!{.p-note}`, `! {.p-note}` and `{.p-note}`

**Reply by schulle4u** [reply]
For me the output is: 

```
<div class="p-note">
<p>Here's a notice with custom class.
Text can span over several lines and contain Markdown text formatting.</p>
</div>
```

**Reply by pftnhr** [reply]
And this is the output of `getContentHtml()`?

**Reply by pftnhr** [reply]
Ah! There must be a line break behind it. However, the `class="p-note"` is not added to the `<p>` tag, which I would prefer at this point.

**Reply by schulle4u** [reply]
Special attributes are currently not supported for paragraphs directly, [at least in MarkdownExtra](https://michelf.ca/projects/php-markdown/extra/#spe-attr). You can use them with headings, fenced code blocks, links and images.

**Reply by pftnhr** [reply]
Ok, too bad, but is there perhaps a way to output the content of the `page.md` without an enclosing `<p>` tag?

**Reply by pftnhr** [reply, marked as answer]
Got it. Took `getContentRaw()` instead of `getContentHtml()` and wrapped a `<p>`-Tag with my desired class around.

**Comment by markseuffert** [comment]
See also datenstrom/community#107.

**Reply by pftnhr** [reply]
Thanks @markseuffert  that's basically what Steffen wrote. I found a solution for myself that does not insert an additional element into the HTML code. See https://github.com/datenstrom/yellow/discussions/934#discussioncomment-8265751

**Reply by markseuffert** [reply]
Thanks Robert. Good to hear you found a solution that works for you. Sometimes it's good to have alternatives. The solution with a HTML wrapper/container that Steffen showed is something we added to the Markdown parser. Out of necessity. If you want a solution without a HTML wrapper/container, this has been previously discussed in Michel Fortin's `michelf/php-markdown` parser. But it's not part of the Markdown Extra syntax or something you can use at the moment.

**Reply by pftnhr** [reply]
In principle, I could also write the text directly into the layout file, but I wanted to retain the editing function. But it's nice to hear that the topic is being discussed.
